flocker/node/agents/blockdevice.py

```diff
@@ -1116,7 +1224,46 @@ def shared_state_changes(self):
         These are the only parts of the state that need to be sent to the
         control service.
         """
-        return (self.node_state, self.nonmanifest_datasets)
+        manifestations = {}
```

Comment:
Actually, I guess `devices` is. But, as far as I can tell `NodeState.devices` is just used to pass data between `discover_state` and `calculate_changes`. I didn't want to get rid of `NodeState.devices` here, but I think that we should eventually do that. Given that, I think keeping the separation we have here seems reasonable.